var/spack/repos/builtin/packages/exaca/package.py

```diff
@@ -40,13 +42,38 @@ class Exaca(CMakePackage):
     depends_on("googletest@1.10:", type="test", when="@1.1:+testing")
     depends_on("kokkos@3.0:", when="@:1.1")
     depends_on("kokkos@3.2:", when="@1.2:")
+    depends_on("kokkos@4.0:", when="@1.3:")
     depends_on("mpi")
     depends_on("nlohmann-json", when="@1.2:")
 
+    for _backend in _kokkos_backends:
+        # Handled separately below
+        if _backend != "cuda" and _backend != "rocm":
+            _backend_dep = "+{0}".format(_backend)
+            depends_on("kokkos {0}".format(_backend_dep), when=_backend_dep)
+
+    for arch in CudaPackage.cuda_arch_values:
+        cuda_dep = "+cuda cuda_arch={0}".format(arch)
+        depends_on("kokkos {0}".format(cuda_dep), when=cuda_dep)
+    for arch in ROCmPackage.amdgpu_targets:
+        rocm_dep = "+rocm amdgpu_target={0}".format(arch)
+        depends_on("kokkos {0}".format(rocm_dep), when=rocm_dep)
```

Comment:
FYI. We are switching from `format()` to f-strings (https://docs.python.org/3/tutorial/inputoutput.html) as we make changes in the core and encourage package developers to do the same.

Reply:
Sounds good - I can go through all the packages I maintain in a follow up